test/test_queue_monkeypatch.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import absolute_import
+
+import unittest
+
+import urllib3
+from urllib3.exceptions import EmptyPoolError
+import Queue
+
+class BadError(Exception):
+    """
+    This should not be raised.
+    """
+    pass
+
+Queue.Empty = BadError
```

Comment:
That's what I tried initially, to import from six and use mock, but I can't get it right, somehow.. the following doesn't raise BadError on unpatched master:
```python
from __future__ import absolute_import

import unittest

import mock

import urllib3
from urllib3.exceptions import EmptyPoolError
from six.moves import queue

class BadError(Exception):
    """ 
    This should not be raised.
    """
    pass


class TestMonkeypatchResistance(unittest.TestCase):
    """ 
    Test that connection pool works even with a monkey patched Queue module,
    see obspy/obspy#1599, kennethreitz/requests#3742, shazow/urllib3#1061.
    """
    def test_queue_monkeypatching(self):
        with mock.patch('queue.Empty', BadError):
            http = urllib3.HTTPConnectionPool(host="localhost", block=True)
            first_conn = http._get_conn(timeout=1)
            with self.assertRaises(EmptyPoolError):
                second_conn = http._get_conn(timeout=1)


if __name__ == '__main__':
    unittest.main()
```